CurrencyExchangeServiceWCF/CurrencyExchangeServiceWCF/Service.svc.cs

```diff
@@ -1,81 +1,100 @@
 ﻿using System;
 using System.Collections.Generic;
+using System.Web.Services.Protocols;
 using System.Text.RegularExpressions;
 using System.Xml;
 
 namespace CurrencyExchangeServiceWCF
 {
-
-    public class Service : IService
+    //souce:        https://www.c-sharpcorner.com/blogs/message-contract-and-its-implementation
+    //              https://www.c-sharpcorner.com/UploadFile/788083/how-to-implement-message-contract-in-wcf/
+    //encription :  https://docs.microsoft.com/en-us/dotnet/framework/wcf/feature-details/using-message-contracts
+    public class Service1 : IService1
     {
-        public string ConvertCurrency(string toCurrency, string value)
+        SoapException soapException ;
+        public Response<string> ConvertCurrency(ConvertCurrenyRequest req)
         {
-            string currencyInputPattern = @"^[a-zA-Z]{3}$";
-            string valueInputPattern = @"^[,.0-9]*$";
+            try
+            {
+                Response<string> r = new Response<string>();
+                if (!req.autHeader.password.Equals("pa$$w0rd") || !req.autHeader.username.Equals("Admin")){
```

Comment:
Also, doseen't this mean the password will be sent without base64 encoding? Or is this done automatically by the framework?

Reply:
see below